python/drned_xmnr/op/base_op.py

```diff
@@ -150,17 +150,20 @@ def setup_drned_env(self, trans):
         env = dict(os.environ)
         root = maagic.get_root(trans)
         drdir = root.drned_xmnr.drned_directory
-        if drdir is None:
+        if drdir == "env":
             try:
                 drdir = env['DRNED']
             except KeyError:
                 raise ActionError('DrNED installation directory not set; ' +
                                   'set /drned-xmnr/drned-directory or the ' +
                                   'environment variable DRNED')
+        elif drdir == "builtin" or drdir is None:
+            env['DRNED'] = os.path.join(self.dev_test_dir, 'drned')
+            drdir = env['DRNED']
         else:
             env['DRNED'] = drdir
         if 'PYTHONPATH' in env:
-            env['PYTHONPATH'] += os.pathsep + drdir
+            env['PYTHONPATH'] += os.pathsep + str(drdir)
```

Comment:
why is this `str()` call necessary, can `drdir` be anything else than a string?

Reply:
Good catch, corrected that mistake left over from some late night debugging.